metriq_gym/job_manager.py

```diff
@@ -0,0 +1,59 @@
+import json
+import uuid
+from typing import List, Dict, Any
+
+DEFAULT_FILE_PATH = ".metric_gym_jobs.jsonl"
+
+# TODO: JobManager is not thread-safe at the moment
+
+
+class JobManager:
+    def __init__(self, file_path: str = DEFAULT_FILE_PATH):
+        self.file_path = file_path
+        self._load_jobs()
+
+    def _load_jobs(self):
+        self.jobs = []
+        try:
+            with open(self.file_path, "r") as file:
+                for line in file:
+                    try:
+                        job = json.loads(line.strip())
+                        self.jobs.append(job)
+                    except json.JSONDecodeError:
+                        continue
+        except FileNotFoundError:
+            self.jobs = []
+
+    def _save_jobs(self):
+        with open(self.file_path, "w") as file:
+            for job in self.jobs:
+                file.write(json.dumps(job, sort_keys=True) + "\n")
+
+    def add_job(self, job: Dict[str, Any]) -> str:
+        job_id = str(uuid.uuid4())
+        job["id"] = job_id
```

Comment:
Should we consider making `job` an object of a class instead of a dictionary or will this be a subsequent PR?